indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -101,7 +112,52 @@ public ForkingTaskRunner(
     this.node = node;
     this.portFinder = new PortFinder(config.getStartPort());
 
-    this.exec = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workerConfig.getCapacity()));
+    this.exec = MoreExecutors.listeningDecorator(
+        Executors.newFixedThreadPool(
+            workerConfig.getCapacity(),
+            new ThreadFactoryBuilder()
+                .setNameFormat("forking-task-runner-%d")
+                .setDaemon(true)
+                .build()
+        )
+    );
+  }
+
+  @Override
+  public List<Pair<Task, ListenableFuture<TaskStatus>>> restore()
```

Comment:
It will try to restore on dirty shutdown too. The graceful shutdown is about trying to get everything in memory persisted to disk, but even if that fails, whatever was persisted to disk can be loaded back up and we will at least have something.